pkg/operator/controller/machineconfig/crio.go

```diff
@@ -32,84 +33,77 @@ import (
 	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 )
 
-const (
-	// CrioProfilingConfigName is the name CRI-O MachineConfig CR
-	CrioProfilingConfigName = "10-crio-nodeobservability"
-
-	// CrioUnixSocketConfFile is the name of the CRI-O config file
-	CrioUnixSocketConfFile = "10-mco-profile-unix-socket.conf"
+// enableCrioProf checks if CRI-O MachineConfig CR for
+// enabling profiling exists, if not creates the resource
+func (r *MachineConfigReconciler) enableCrioProf(ctx context.Context) (bool, error) {
+	createdNow := false
+	namespace := types.NamespacedName{Name: CrioProfilingConfigName}
 
-	// CrioServiceFile is the name of the CRI-O systemd service unit
-	CrioServiceFile = "crio.service"
+	err := r.createCrioProfConf(ctx)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return createdNow, err
+	}
+	if err == nil {
+		createdNow = true
+	}
 
-	// CrioUnixSocketConfData contains the configuration required
-	// for enabling CRI-O profiling
-	CrioUnixSocketConfData = `[Service]
-Environment="ENABLE_PROFILE_UNIX_SOCKET=true"`
-)
+	// grace time for client cache to get refreshed
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Why do you think this is needed?

Reply:
At certain times observed client cache not getting updated and immediate fetch was failing, adding delay here solved that.

Reply:
I think you have a fairly up-to-date obj already:

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/typed_client.go#L51